src/js/net.js

```diff
@@ -279,7 +279,11 @@ function connect(socket, ip, port) {
     }
   };
 
-  socket._handle.connect(ip, port, afterConnect);
+  var err = socket._handle.connect(ip, port, afterConnect);
+  if (err) {
+    emitError(socket, new Error('connect failed - status: ' +
+      TCP.errname(err)));
+  }
```

Comment:
@vprunet You mean 2 local uses should be deferred? Then I am worried the uses of the `emit*` outside this module like `src/js/stream_writable.js` is a bit different from here. The rests uses propagate event instantly, not deferred.